src/Decode.fs

```diff
@@ -1150,7 +1157,7 @@ If you can't use one of these types, please pass an extra decoder.
                 // Disable seq support because I don't know how to implement it on Thoth.Json.Net side
                 // elif fullname = typedefof<obj seq>.FullName then
                 //     t.GenericTypeArguments.[0] |> (autoDecoder extra caseStrategy false) |> seq |> boxDecoder
-                elif fullname = typedefof< Map<string, obj> >.FullName then
+                elif fullname = typedefof< Map<System.IComparable, obj> >.FullName then
```

Comment:
Note for myself, check if the decoder is compatible with previous JSON representation